.github/workflows/continuous-integration.yml

```diff
@@ -6,28 +6,66 @@ on:
 
 jobs:
   static-analysis-psalm:
-    name: Static Analysis with Psalm
-    runs-on: ubuntu-latest
+    name: "Static Analysis with Psalm"
+    runs-on: "ubuntu-latest"
+
+    strategy:
+      matrix:
+        php-version:
+          - "7.4"
 
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v2
+      - name: "Checkout code"
+        uses: "actions/checkout@v2"
 
       - name: "Install PHP"
-        uses: "shivammathur/setup-php@1.8.1"
+        uses: "shivammathur/setup-php@v2"
         with:
           coverage: "none"
-          php-version: "7.4"
+          php-version: "${{ matrix.php-version }}"
 
       - name: "Cache dependencies installed with composer"
-        uses: "actions/cache@v1.0.3"
+        uses: "actions/cache@v1"
         with:
           path: "~/.composer/cache"
-          key: "composer-${{ hashFiles('composer.lock') }}"
-          restore-keys: "composer-"
+          key: "php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock') }}"
```

Comment:
This looks like an unnecessary complication. What's the scenario where we're going to run the code style check using different PHP versions?

Reply:
Could you point me to the thread, please? I could find the discussion regarding the build matrix.

Reply:
https://github.com/doctrine/orm/pull/8116#discussion_r418949951

Reply:
sorry, let me elaborate, because that link doesn't help you.

I think, even if the matrix has a single php version, it enables us to use the same `key` everywhere, which helps cache hits. rather than maintaining 
`key: "php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock') }}"` where a matrix is used and
`key: "php-7.4-composer-locked-${{ hashFiles('composer.lock') }}"`
where a matrix is not used.

Reply:
So, by “everywhere” you mean that if named consistently, the same cache entry can be shared between multiple actions? Makes sense in this case.

Reply:
Multiple action in the same repo yes. Thanks for phrasing it better!